### PR TITLE
[DEV APPROVED] 7923 global nav add tests

### PR DESCRIPTION
**Pull request by davidtrussler**
This PR adds Karma tests for the Global Nav in the following areas: 

* Test for initialisation of component: check init is calling the appropriate methods. This also slightly refactors GlobalNav.js to put the two one-liners into a separate method. It's is a bit neater I think and helps test the init as a single unit which is probably better practice.
* Tests for binding events. In fact this just tests that the uninitialised class is removed when the component loads
* Test for mobile animation. This should test that the no-transition class is added/removed when the window is resized. I have so far been unable to simulate the resize event in this test so this is left as an empty test (i.e. it passes but only because it isn’t doing anything but can serve as a reminder to work on it at a later date).
* Tests for mobile interaction. This tests: 
    * the nav visibility toggles when the menu button is clicked
    * the dropdown visibility toggles when the clump heading is clicked
    * the dropdown visibility toggles when the subnav heading is clicked
    * the nav is closed when the close button is clicked
    * the nav is closed when the overlay is clicked
* Tests for desktop interaction This tests: 
    * the dropdown is closed on a touch event outside of global nav
    * the dropdown is closed when the user moves the mouse outside of the global nav
    * the dropdown is closed when the user hovers on a clump heading
    * the dropdown is opened when the user touches a clump heading
* Tests for keyboard interactions. This tests: 
    * the enter key opens the dropdown and moves focus into this
    * the spacebar key opens dropdown
    * when at the top level the left arrow key moves focus to the previous element on the top level or wraps if on the first element
    * when at the top level the right arrow key moves focus to the next element on top level or wraps if on the last element
    * when at the top level the down arrow key opens the dropdown
    * the escape key closes dropdown and moves focus back to top level
    * the up arrow key moves focus to the previous element or wraps to the last element if on the first
    * when at the secondary level the down arrow key moves focus to the next element or wraps to the first element if on the last
    * when at the secondary level the left arrow key closes the dropdown and moves focus to the previous top level element or the last element if already on the first
    * when at the secondary level the right arrow key closes the dropdown and moves focus to the next top level element or the first element if already on the last

---
This change is [Reviewable](https://reviewable.io/reviews/moneyadviceservice/frontend/1719)

### CODE REVIEW

**Comment by johnplayer1982** [comment]
Very nice!! Tested locally, broke it intentionally to test the fixture, looks good! :+1:

Lets talk testing focus in the next FE team meeting

**Comment by davidtrussler** [comment]
Have added a further commit which sorts out the focus issues on the keyboard events

File: spec/javascripts/tests/GlobalNav_spec.js [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+    });
+  });
+
+  describe('Mobile animation', function() {
+    beforeEach(function() {
+      this.obj.init();
+    });
+
+    it('toggles the no-transition class when viewport size changes', function() {
+      // This is not functional because I have been unable to simulate the resize event - DT
```
Comment (tomas-stefano) [inline spec/javascripts/tests/GlobalNav_spec.js]: Is it possible to add a xit for pending? What is the intended to treat this in the future?

**Comment by tomas-stefano** [comment]
Just a minor comment but the code looks good 👍 Very very nice!!!

File: spec/javascripts/tests/GlobalNav_spec.js (hunk 1 above)
Comment (norm) [inline spec/javascripts/tests/GlobalNav_spec.js]: `window.dispatchEvent(new Event('resize'));` ?

Comment (davidtrussler) [inline spec/javascripts/tests/GlobalNav_spec.js]: @tomas-stefano I'm not sure about how to deal with this and ensure it gets done some time. 

@norm The problem with the resize event is that the component uses the media query class that is added to the <html> tag but this doesn't get read in the test scenario.

**Pull request merged**